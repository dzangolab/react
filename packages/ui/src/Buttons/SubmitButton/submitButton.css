.button-submit {
  align-items: center;
  background: var(--button-submit-bg-color, #1c4786);
  border-radius: 6px;
  border: none;
  color: var(--button-submit-color, #ffffff);
  cursor: pointer;
  display: flex;
  font-size: 1rem;
  justify-content: center;
  padding: 1rem 0.625rem;
}

.button-submit:disabled {
  background: #dddddd;
}
