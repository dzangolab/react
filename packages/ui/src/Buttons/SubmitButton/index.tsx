import React from "react";

import LoadingIcon from "../../LoadingIcon";

import "./submitButton.css";

interface Properties {
  disabled?: boolean;
  label: string;
  loading?: boolean;
}

const SubmitButton = ({ disabled, label, loading }: Properties) => {
  return (
    <button
      type="submit"
      className="button-submit"
      disabled={disabled || loading}
    >
      {label}

      {loading ? <LoadingIcon /> : ""}
    </button>
  );
};

export default SubmitButton;
