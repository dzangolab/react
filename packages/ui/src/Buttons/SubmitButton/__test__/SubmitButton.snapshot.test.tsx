import React from "react";
import renderer, { create } from "react-test-renderer";
import { expect, test } from "vitest";

import SubmitButton from "..";

function toJson(component: renderer.ReactTestRenderer) {
  const result = component.toJSON();

  expect(result).toBeDefined();
  expect(result).not.toBeInstanceOf(Array);

  return result as renderer.ReactTestRendererJSON;
}

test("Component matches snapshot", () => {
  const component = create(<SubmitButton label="Submit" />);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
});
