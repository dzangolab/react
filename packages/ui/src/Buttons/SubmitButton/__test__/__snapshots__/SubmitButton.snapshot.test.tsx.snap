// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Component matches snapshot 1`] = `
<button
  className="dz-button primary filled medium"
  type="submit"
>
  <span
    className="label"
  >
    Submit
  </span>
</button>
`;
